app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTablePresenter.java

```diff
@@ -61,31 +60,30 @@ public void init(String orgUnitUid, String periodTypeName, String catCombo,
                 Flowable.zip(
                         tableRepository.getDataSet(),
                         tableRepository.getCatComboName(catCombo),
-                        Pair::create
+                        tableRepository.getPeriod(),
+                        Trio::create
                 )
                         .subscribeOn(schedulerProvider.io())
                         .observeOn(schedulerProvider.ui())
                         .subscribe(
-                                data -> view.renderDetails(data.val0(), data.val1()),
+                                data -> view.renderDetails(data.val0(), data.val1(), data.val2()),
                                 Timber::e
                         )
         );
 
         disposable.add(
-                tableRepository.dataSetStatus()
+                view.observeSaveButtonClicks()
+                        .subscribeOn(schedulerProvider.ui())
+                        .toFlowable(BackpressureStrategy.LATEST)
+                        .debounce(500, TimeUnit.MILLISECONDS, schedulerProvider.io())
+                        .switchMap(o -> tableRepository.completeDataSetInstance())
```

Comment:
do you want to prevent fast clicks and get the last one? I think this could be valid too :+1: ? https://github.com/ReactiveX/RxJava/wiki/Filtering-Observables#throttlefirst-example

Reply:
I like it! :)

Reply:
i think your solution is okay. I saw this others variant using Missing
observable.toFlowable(BackpressureStrategy.MISSING).debounce(1000,TimeUnit.MILLISECONDS)